api/src/main/java/de/oliver/fancyholograms/api/Hologram.java

```diff
@@ -185,10 +186,11 @@ public final Component getShownText(@Nullable final Player player) {
         if (!(getData().getTypeData() instanceof TextHologramData textData)) {
             return null;
         }
-
-        var text = String.join("\n", textData.getText());
-
-        return ModernChatColorHandler.translate(text, player);
+        var text = textData.getText();
+        if (FloodgateChecker.isBedrockPlayer(player) && !textData.getBedrockText().isEmpty()) {
```

Comment:
This code would currently break for servers that do not have floodgate installed, it could be worth implementing FloodgateChecker as an object that is enabled onEnable and then accessed through a `getFloodgateChecker` method

Reply:
> This code would currently break for servers that do not have floodgate installed, it could be worth implementing FloodgateChecker as an object that is enabled onEnable and then accessed through a `getFloodgateChecker` method

i believe this won't break, as the floodgate checker class is our own. then we use a dirty try/catch to check if the class exists in the class loader. if it errors then they don't have floodgate. this is a very dirty way and your suggestion is definitely better